src/App.js

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react'
+import { Routes, Route, createSearchParams, useSearchParams, useNavigate } from "react-router-dom"
+import { useDispatch, useSelector } from 'react-redux'
+import 'reactjs-popup/dist/index.css'
+import { fetchMovies } from './data/moviesSlice'
+import { ENDPOINT_SEARCH, ENDPOINT_DISCOVER, ENDPOINT, API_KEY } from './constants'
+import Header from './components/Header'
+import Movies from './components/Movies'
+import Starred from './components/Starred'
+import WatchLater from './components/WatchLater'
+import YouTubePlayer from './components/YoutubePlayer'
+import './app.scss'
+
+const App = () => {
+
+  const state = useSelector((state) => state)
```

Comment:
We should use movies selector instead of all application state
```suggestion
   const movies = useSelector((state) => state.movies)
```

Reply:
it can be also like 
const {movies} = useSelector(state => state);

Reply:
if you get the entire state the component will be re-rendered when any prop of the state change, we should select only the reducer needed